app/assets/stylesheets/common/float-kit/d-default-toast.scss

```diff
@@ -70,6 +70,18 @@
     min-height: 30px;
   }
 
+  &__progress-bar {
+    height: 5px;
+    top: 0;
+    left: 0;
+    position: absolute;
+    background-color: var(--success);
+  }
```

Comment:
We should add: `will-change: width;` so that will force the element to have its own compositor layer, we shouldn't abuse it as it will use more memory, but in this case it should be fine

Reply:
I knew you might say this :p

I don't think there's any problem with code really, that's maybe a problem if you animate dozens of things in the screen. Especially in this case, as it will indeed change as soon as the element is added.

Reply:
We talked of this with @davidtaylorhq, he convinced me :p

Reply:
I've reworked this to make it more performant. Actually it turns out to be much less complex using js animate as we don't need to handle any calculations at all or do any loops or setInterval.

I mentioned to @jjaffeux yesterday that I wanted to refactor this to move the handling to the d-toasts component as the progress bar will only be used when autoClose is enabled, so it makes sense to have this logic within the same component to simplify things.

I demo'd this in the APAC call earlier after switching to transform scaleX and js animate, although the one observation was that using `animation.pause` was a bit jarring when we mouse out (as the timer restarts and therefore width of progress bar jumps to full again). I've simplified it further to just cancel the animation when using mouseover, and then creating a new animation when starting again.